piksi_tools/console/settings_view.py

```diff
@@ -212,6 +227,40 @@ def _factory_default_button_fired(self):
                         + "Are you sure you want to reset to factory defaults?"
     confirm_prompt.run(block=False)
 
+  def _settings_save_to_file_fired(self):
+    dialog = FileDialog(label='Choose Location to save settings file',
+                        action='save as', wildcard=self.file_wildcard)
+    dialog.open()
+    if dialog.return_code == OK:
+      filepath = os.path.join(dialog.directory, dialog.filename)
+      self.save_settings_to_file(filepath)
+    else:
+      print 'Unable to save file'
+
+  def save_settings_to_file(self, filepath):
+    fio = FileIO(self.link)
+    with open(filepath, 'w') as config_file:
+      blob = fio.read("config")
+      config_file.write(blob)
+
+  def load_settings_from_file(self, filepath):
+    fio = FileIO(self.link)
+    with open(filepath, 'r') as config_file:
+      blob = config_file.read()
+      fio.write("config", blob)
+    self.link.send(SBP_MSG_RESET, '')
+    self._settings_read_button_fired()
```

Comment:
Caling `_settings_read_button_fired()` happens a couple other places too, but is kind of janky - can we refactor so other code that wants to trigger a settings read doesn't need to use the GUI hooks?